db/migrate/20180111162442_move_pictures_blobs_to_pictures.rb

```diff
@@ -0,0 +1,51 @@
+class MovePicturesBlobsToPictures < ActiveRecord::Migration[5.0]
+  class BinaryBlob < ActiveRecord::Base
+    has_many :binary_blob_parts, -> { order(:id) }, :dependent => :delete_all
+    belongs_to :resource, :polymorphic => true
+    def data
+      binary_blob_parts.pluck(:data).join
+    end
+  end
+  class BinaryBlobPart < ActiveRecord::Base; end
+  class Picture < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+  end
+
+  def up
+    add_column :pictures, :content, :binary
+    add_column :pictures, :extension, :string
+    add_column :pictures, :md5, :string
+
+    say_with_time("Moving picture content from BinaryBlobs to the pictures table") do
+      BinaryBlob.where(:resource_type => "Picture").all.each do |blob|
+        Picture.update(blob.resource_id, :content => blob.data, :extension => blob.data_type, :md5 => blob.md5)
+        blob.destroy
+      end
+    end
+  end
+
+  def down
+    say_with_time("Moving picture content from the pictures table to BinaryBlobs") do
+      Picture.in_my_region.all.each do |picture|
+        size = picture.content.try(:length).to_i
+
+        BinaryBlob.create!(
```

Comment:
We shouldn't create the binaryblob if the size is 0.  Though that seems like a real edge case.